experiment/tunnel_community/hidden_peer_discovery.py

```diff
@@ -0,0 +1,72 @@
+import asyncio
+import os
+import time
+from binascii import hexlify, unhexlify
+from pathlib import Path
+
+from ipv8.taskmanager import TaskManager
+
+from experiment.tool.tiny_tribler_service import TinyTriblerService
+
+EXPERIMENT_RUN_TIME = int(os.environ.get('EXPERIMENT_RUN_TIME', 3600 * 3))
+
+
+class Service(TinyTriblerService, TaskManager):
+    def __init__(self, working_dir, config_path):
+        super().__init__(Service.create_config(working_dir, config_path), None, working_dir, config_path)
+        TaskManager.__init__(self)
+        self.swarm = None
+        self.start = time.time()
+        self.results = []
+        self.register_task('monitor_swarm', self.monitor_swarm, interval=5)
+        self.register_task('_graceful_shutdown', self._graceful_shutdown, delay=EXPERIMENT_RUN_TIME)
+
+    @staticmethod
+    def create_config(working_dir, config_path):
+        config = TinyTriblerService.create_default_config(working_dir, config_path)
+        config.set_ipv8_enabled(True)
+        config.set_tunnel_community_enabled(True)
+        config.set_dht_enabled(True)
+        config.set_torrent_checking_enabled(False)
+        return config
+
+    def _graceful_shutdown(self):
+        task = asyncio.create_task(self.on_tribler_shutdown())
+        task.add_done_callback(lambda result: TinyTriblerService._graceful_shutdown(self))
+
+    async def on_tribler_shutdown(self):
+        await self.shutdown_task_manager()
+        with open('hidden_peer_discovery.txt', 'w') as f:
+            f.write("Time Seeders Introduction-points Connections Partial-connections Last-lookup Last-DHT-lookup\n")
+            for result in self.results:
+                f.write(' '.join(map(str, result)) + '\n')
+
+    async def on_tribler_started(self):
+        info_hash = unhexlify('e24d8e65a329a59b41a532ebd4eb4a3db7cb291b')
+        self.session.tunnel_community.join_swarm(info_hash, 1, seeding=False)
+        self.swarm = self.session.tunnel_community.swarms[info_hash]
+        print(f'Joining hidden swarm {hexlify(info_hash)}')
```

Comment:
Could you add a `# noqa: T001` to this line so the linter doesn't error out on it next time it sees this file?

Reply:
Sure